source/MaterialXGenShader/ColorManagementSystem.h

```diff
@@ -0,0 +1,105 @@
+#ifndef MATERIALX_COLOR_MANAGEMENT_SYSTEM_H
+#define MATERIALX_COLOR_MANAGEMENT_SYSTEM_H
+
+#include <MaterialXCore/Document.h>
+#include <MaterialXCore/Library.h>
+
+#include <MaterialXGenShader/Factory.h>
+#include <MaterialXGenShader/ShaderNode.h>
+#include <MaterialXGenShader/ShaderNodeImpl.h>
+#include <MaterialXGenShader/TypeDesc.h>
+
+namespace MaterialX
+{
+
+class ShaderGenerator;
+
+/// A shared pointer to a ColorManagementSystem
+using ColorManagementSystemPtr = shared_ptr<class ColorManagementSystem>;
+
+/// @struct @ColorSpaceTransform
+/// Structure that represents color space transform information
+struct ColorSpaceTransform
+{
+    ColorSpaceTransform(const string& ss, const string& ts, const TypeDesc& t);
+
+    string sourceSpace;
+    string targetSpace;
+    const TypeDesc* type;
+
+    /// Comparison operator
+    bool operator==(const ColorSpaceTransform &other) const
+    {
+        return sourceSpace == other.sourceSpace &&
+               targetSpace == other.targetSpace &&
+               type == other.type;
+    }
+};
+
+/// @struct @ColorSpaceTransformHash
+/// ColorSpaceTransform hash function
+struct ColorSpaceTransformHash {
+    size_t operator()(const ColorSpaceTransform &transform ) const
+    {
+        if (transform.type == nullptr)
+        {
+            return std::hash<string>()(transform.sourceSpace + transform.targetSpace);
+        }
+        else
+        {
+            return std::hash<string>()(transform.sourceSpace + transform.targetSpace + transform.type->getName());
+        }
+    }
+};
+
+/// @class @ColorManagementSystem
+/// Abstract base class for a ColorManagementSystem.
+///
+class ColorManagementSystem
+{
+  public:
+    /// Return the ColorManagementSystem name
+    virtual const string& getName() const = 0;
+
+    /// Return the ColorManagementSystem configFile
+    const string& getConfigFile() const
+    {
+        return _configFile;
+    }
+
+    /// Sets the config file
+    void setConfigFile(const string& configFile);
+
+    /// Load the library of implementations from the provided document
+    virtual void loadLibrary(DocumentPtr document);
+
+    /// Create a node to use to perform the given color space transformation.
+    ShaderNodePtr createNode(const ColorSpaceTransform& transform, const string& prefix);
```

Comment:
Is this prefix for making the node names unique?

Would it make sense to let the client handle the full name instead and so change from `prefix` to `name` here, and then also remove the getShaderNodeName() method. I think the client side is best at determining which names are unique or in use, and the node name itself doesn't really matter as long as it's unique.

Reply:
When we use this in ShaderX we could for example just name the transform nodes "nodename_portname_cm", to make them unique since "nodename_portname" would already be unique.

Reply:
Maybe it's the name `prefix` I have most problems with. Right now it looks like an optional prefix you can set on your nodes. But if you leave this empty you will get node name collisions.
So maybe change it to `name` and you can still append the transform space strings if you like. Although the caller can also append such information into the name, and the caller can then decide if it's needed or not.

Reply:
Ok